### PR TITLE
Issue#300: Add pagination to challenge list page

### LINKED ISSUE #300: Add pagination to challenge List page
Need to add pagination for each challenge list (Past, Ongoing & Future) on challenge page.

For pagination use load more, clicking on which gets next list of data.

Let me know if anyone finds it difficult :)

### PR DESCRIPTION
**Pull request by Obtin**
Fix #300 Challenge list page now has load more(active when more than 3 challenges available) and load less (when more than 3 appear on screen) feature.
pagination3

### CODE REVIEW

**Comment by RishabhJain2018** [comment]
@Obtin Can you please check as the PR doesn't work on my local system.
Also, I think you can place a circular button at the right side of the list of three challenges. What do you say @aka-jain ?

**Comment by Obtin** [comment]
@RishabhJain2018, The PR works fine on my system. Please check again

**Comment by aka-jain** [comment]
@Obtin for pagination please refer the design from other component like teams card. We have left and right button to navigate in team card. Let us know if you can't find it or have any doudt regarding it's design.

File: frontend/src/js/controllers/challengeListCtrl.js [hunk 1]
```diff
@@ -53,6 +53,19 @@
...
                     utilities.storeData("challengeCreator", vm.challengeCreator);
                 }
+                if(vm.currentList.length !=0){
+                     vm.presentLoadIndex=3;
+                     vm.showMore = function() {
+                        if (vm.presentLoadIndex < vm.currentList.length){
+                           vm.presentLoadIndex = vm.presentLoadIndex+1;
+                        }
+                     };
+                     vm.showLess = function() {
```
Comment (aka-jain) [inline frontend/src/js/controllers/challengeListCtrl.js]: Why there is a check of 3 @Obtin, we are already sending pagination limit in API. Please check response.

File: frontend/src/views/web/challenge/challenge-list.html [hunk 2]
```diff
@@ -45,11 +50,16 @@
             </a>
         </div>
     </div>
+   <!-- load more or less for upcoming challenges -->
+    <div>
```
Comment (aka-jain) [inline frontend/src/views/web/challenge/challenge-list.html]: Please refer team card for pagination logic.

File: frontend/src/js/controllers/challengeListCtrl.js (hunk 1 above)
Comment (Obtin) [reply frontend/src/js/controllers/challengeListCtrl.js]: Done

**Comment by Obtin** [comment]
@RishabhJain2018 @aka-jain I guess this was missed. Please have a look at it and suggest me the required changes

**Comment by deshraj** [comment]
Closing this due to inactivity. @Obtin please feel free to reopen it if you are still working on it.

**Pull request closed without merging**